code/ATMOSPHERICS/pipe/construction.dm

```diff
@@ -100,8 +102,12 @@ var/global/list/heat_pipes = list(PIPE_HE_STRAIGHT, PIPE_HE_BENT, PIPE_JUNCTION,
 	qdel(src)
 	return 2
 var/list/bent_dirs = list(NORTH|SOUTH, WEST|EAST)
-/obj/item/pipe/New(var/loc, var/pipe_type as num, var/dir as num, var/obj/machinery/atmospherics/make_from = null)
+/obj/item/pipe/New(var/loc, var/pipe_type as num, var/dir as num, var/obj/machinery/atmospherics/make_from = null, frequency, id)
 	..()
+	if (frequency)
+		src.frequency = frequency
```

Comment:
as with elsewhere, there is no need for these `src.` prefixes

Reply:
I think in this case it's needed ? `frequency` is gonna be the parameter passed, `src.frequency` is the datum's variable.

Reply:
Good point; this is more of an argument for improving names of... arguments